icu4c/source/i18n/number_longnames.cpp

```diff
@@ -265,6 +265,20 @@ UnicodeString LongNameHandler::getUnitDisplayName(
     return simpleFormats[DNAM_INDEX];
 }
 
+UnicodeString LongNameHandler::getUnitPattern(
+        const Locale& loc,
+        const MeasureUnit& unit,
+        UNumberUnitWidth width,
+        StandardPlural::Form pluralForm,
+        UErrorCode& status) {
+    if (U_FAILURE(status)) {
+        return ICU_Utility::makeBogusString();
+    }
+    UnicodeString simpleFormats[ARRAY_LENGTH];
+    getMeasureData(loc, unit, width, simpleFormats, status);
+    return simpleFormats[pluralForm];
```

Comment:
This function may get called twice, and each time you fetch strings for all forms an return one of them. How about instead taking two UnicodeString's by reference and assigning to them the pluralForm and the OTHER form (e.g., pattern / otherPattern)? The call site would then not need to call this again for OTHER. (Return a boolean = U_SUCCESS)